cumulusci/robotframework/Salesforce.py

```diff
@@ -1131,3 +1124,58 @@ def breakpoint(self):
         set, this keyword will have no effect on a running test.
         """
         return None
+
+    def _check_for_classic(self):
+        """Switch to lightning if we land on a classic page
+
+        This seems to happen randomly, causing tests to fail
+        catastrophically. The idea is to detect such a case and
+        auto-click the "switch to lightning" link
+
+        """
+        try:
+            # we don't actually want to wait here, but if we don't
+            # explicitly wait, we'll implicitly wait longer than
+            # necessary.  This needs to be a quick-ish check.
+            self.selenium.wait_until_element_is_visible(
+                "class:switch-to-lightning", timeout=2
+            )
```

Comment:
2 seconds is still not quick if it has to happen at the start of every test even when things are working properly. Can we invert the check so that we're looking for something that is expected to be on every lightning page (maybe the #auraLoadingBox) and only look for the switch-to-lightning link if that fails?

Reply:
We're already doing that check. We only get to this section if we didn't find a specific lightning component. By the time this code runs, we've already waited for `document.readyState`, waited for aura, and waited for a lightning locator.